src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.UnifiedBuild.Tasks/JoinVerticals.cs

```diff
@@ -23,80 +22,40 @@ public class JoinVerticals : Microsoft.Build.Utilities.Task
-    /// </summary>
-    [Required]
-    public required string MainVertical { get; init; }
-
-    /// <summary>
-    /// Azure DevOps build id
-    /// </summary>
-    [Required]
-    public required string BuildId { get; init; }
-
-    /// <summary>
-    /// Azure DevOps token, required scopes: "Build (read)", allowed to be empty when running in a public project
-    /// </summary>
-    public string? AzureDevOpsToken { get; set; }
-
-    /// <summary>
-    /// Azure DevOps organization
-    /// </summary>
-    [Required]
-    public required string AzureDevOpsBaseUri { get; init; }
-
-    /// <summary>
-    /// Azure DevOps project
-    /// </summary>
-    [Required]
-    public required string AzureDevOpsProject { get; init; }
-
     /// <summary>
     /// Location of downloaded artifacts from the main vertical
     /// </summary>
     [Required]
-    public required string MainVerticalArtifactsFolder { get; init; }
+    public required string VerticalArtifactsBaseFolder { get; init; }
 
     /// <summary>
     /// Folder where packages and assets will be stored
     /// </summary>
     [Required]
     public required string OutputFolder { get; init; }
 
-
-    private const string _artifactNameSuffix = "_Artifacts";
     private const string _assetsFolderName = "assets";
     private const string _packagesFolderName = "packages";
-    private const int _retryCount = 10;
+    private const string _releaseFolderName = "Release";
 
     public override bool Execute()
-    {
-        ExecuteAsync().GetAwaiter().GetResult();
-        return !Log.HasLoggedErrors;
-    }
-
-    private async Task ExecuteAsync()
     {
         List<BuildAssetsManifest> manifests = VerticalManifest.Select(xmlPath => BuildAssetsManifest.LoadFromFile(xmlPath.ItemSpec)).ToList();
         BuildAssetsManifest mainVerticalManifest = manifests
-            .FirstOrDefault(manifest => StringComparer.OrdinalIgnoreCase.Equals(manifest.VerticalName, MainVertical))
-            ?? throw new ArgumentException($"Couldn't find main vertical manifest {MainVertical} in vertical manifest list");
-
-        string mainVerticalName = mainVerticalManifest.VerticalName!;
+            .FirstOrDefault()
+            ?? throw new ArgumentException($"Vertical manifest list was empty");
 
-        JoinVerticalsAssetSelector joinVerticalsAssetSelector = new JoinVerticalsAssetSelector();
+        JoinVerticalsAssetSelector joinVerticalsAssetSelector = new();
 
         List<AssetVerticalMatchResult> selectedVerticals = joinVerticalsAssetSelector.SelectAssetMatchingVertical(manifests).ToList();
 
         var notMatchedAssets = selectedVerticals.Where(o => o.MatchType == AssetVerticalMatchType.NotSpecified).ToList();
-        Log.LogMessage(MessageImportance.High, $"### {notMatchedAssets.Count()} Assets not properly matched to vertical: ###");
+        Log.LogMessage(MessageImportance.High, $"### {notMatchedAssets.Count} Assets not properly matched to vertical: ###");
```

Comment:
Is it possible to turn this into an error now?

Reply:
Yes, I'll do that